### PR TITLE
[2/5] Visualizations (merge after 1/5)

### PR DESCRIPTION
**Pull request by ProfFan**

### CODE REVIEW

**Comment by dellaert** [comment]
I’ll review after 1/5 is merged as this PR seems cumulative

File: Sources/BeeTracking/Visualizations.swift [hunk 1]
```diff
@@ -69,4 +69,106 @@ public func plotPatchWithGT(frame: Tensor<Float>, actual: Pose2, expected: Pose2
   ).makeNumpyArray() / 255.0)
   ax[1].title.set_text("Labeling")
   return (fig, ax)
-}
\ No newline at end of file
+}
+
+public func errorPlaneTranslation<
```
Comment (dellaert) [inline Sources/BeeTracking/Visualizations.swift:74]: document, and all arguments

Comment (dellaert) [reply Sources/BeeTracking/Visualizations.swift:74]: Maybe even make LikelihoodModel right away :-) Or better: make an extension for TrackingLikelihoodModel?

File: Sources/BeeTracking/Visualizations.swift [hunk 2]
```diff
@@ -69,4 +69,106 @@ public func plotPatchWithGT(frame: Tensor<Float>, actual: Pose2, expected: Pose2
...
+      /// x is horiz movement, but is vertical dim in imshow
+      fg[j, i] = Tensor(fg_nll)
+      bg[j, i] = Tensor(bg_nll)
+      errors[j, i] = Tensor(e)
+    }
+  }
+  return (fg, bg, errors)
+}
+
+public func plotErrorPlaneTranslation<
```
Comment (dellaert) [inline Sources/BeeTracking/Visualizations.swift:119]: document, extension for TrackingLikelihoodModel?

File: Sources/BeeTracking/Visualizations.swift [hunk 3]
```diff
@@ -69,4 +69,106 @@ public func plotPatchWithGT(frame: Tensor<Float>, actual: Pose2, expected: Pose2
...
+    at: pose,
+    deltaXs: deltaXs,
+    deltaYs: deltaYs,
+    statistics: statistics,
+    encoder: encoder,
+    foregroundModel: foregroundModel,
+    backgroundModel: backgroundModel
+  )
+
+  // let trans_mins = [e, fg, bg].map { $0.min() }
```
Comment (dellaert) [inline Sources/BeeTracking/Visualizations.swift:145]: make a flagged option

File: Sources/BeeTracking/Visualizations.swift [hunk 4]
```diff
@@ -69,4 +69,106 @@ public func plotPatchWithGT(frame: Tensor<Float>, actual: Pose2, expected: Pose2
...
+  // let targetSize = (40, 70)
+  let (fig, axs) = plt.subplots(2, 2, figsize: Python.tuple([12, 10])).tuple2
+  let img_m = axs[0][0].imshow(frame.patch(
+    at: OrientedBoundingBox(center: pose, rows: 40 + 20 * 2, cols: 70 + 20 * 2)
+  ).makeNumpyArray() / 255.0, cmap: "gray")
+  fig.colorbar(img_m, ax: axs[0][0])
+  axs[0][0].title.set_text("Image")
+  axs[0][0].set(xlabel: "x displacement", ylabel: "y displacement")
+
+  let fg_m = axs[0][1].imshow(fg.makeNumpyArray(), cmap: "hot", interpolation: "nearest") // , vmin: trans_min, vmax: trans_max)
```
Comment (dellaert) [inline Sources/BeeTracking/Visualizations.swift:158]: document blocks

**Review by dellaert — APPROVED** [review]

File: Sources/BeeTracking/Visualizations.swift (hunk 1 above)
Comment (ProfFan) [reply Sources/BeeTracking/Visualizations.swift:74]: Yeah I think I can try make that happen before merging

File: Sources/BeeTracking/Visualizations.swift (hunk 4 above)
Comment (ProfFan) [reply Sources/BeeTracking/Visualizations.swift]: done

File: Sources/BeeTracking/Visualizations.swift (hunk 3 above)
Comment (ProfFan) [reply Sources/BeeTracking/Visualizations.swift]: done

**Comment by ProfFan** [comment]
I'll merge to unblock :)

**Pull request merged**